src/main/java/jenkins/scm/api/SCMNavigatorDescriptor.java

```diff
@@ -40,6 +40,12 @@ protected SCMNavigatorDescriptor(Class<? extends SCMNavigator> clazz) {
         super(clazz);
     }
 
+    public abstract String getDescription();
+
+    public abstract String getCategoryId();
+
+    public abstract String getIconFilePathPattern();
```

Comment:
:bug: (3×) for introducing an `abstract` method. Should make it concrete and return null by default.